src/scan_find.cpp

```diff
@@ -43,6 +46,41 @@ namespace {
     }
 }
 
+void scan_find_sbuf(scanner_params &sp, sbuf_t &sbuf)
+{
+    feature_recorder &f = sp.named_feature_recorder("find");
+
+    auto *tbuf = sbuf_t::sbuf_malloc(sp.sbuf->pos0, sp.sbuf->bufsize+1, sp.sbuf->bufsize+1);
+    memcpy(tbuf->malloc_buf(), sp.sbuf->get_buf(), sp.sbuf->bufsize);
```

Comment:
are you doing the `memcpy` solely so you can null-terminate it?

Reply:
Yes.